Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -41,11 +41,24 @@ public struct GraphQLHTTPResponseError: Error, LocalizedError {
   }
 }
 
+/// Used to intercept the http requests/errors/responses. This allows modifying url requests dynamically without recreating transports
+public protocol HTTPNetworkTransportDelegate: class {
```

Comment:
I'm using this directly on the application level to verify if this works for various use cases.
Question: what is the reason for this protocol - it looks like the only responsibility is to create interception for each request.